python/tk_multi_workfiles/step_list_filter.py

```diff
@@ -58,12 +61,52 @@ def get_filter_from_filter_list(step_list):
     return step_filter
 
 
+def get_steps_for_entity_type(entity_type):
+    """
+    Return cached Flow Production Tracking (ShotGrid) Step dictionaries for a given Entity type.
+
+    The list is sourced from the framework's schema cache
+    (`CachedShotgunSchema.get_pipeline_steps_by_entity_type`) and respects the
+    `project_visible_pipeline_steps` app setting and the current project context.
+    When project visibility is enabled, only Steps exposed via project schema
+    'step_*' fields are included.
+
+    :param str entity_type: A Shotgun Entity type (e.g. 'Asset', 'Shot').
+    :returns: List of Step dictionaries with at least 'id', 'code', 'entity_type'
+              and optionally 'color'. Returns an empty list if none are available.
+    """
+    return list(get_step_map().get(entity_type, []))
+
+
+def get_step_map():
```

Comment:
should we make this method private and only expose `get_steps_for_entity_type`?
I feel like `get_step_map` is only intended to be used in `StepListWidget`
Also, I'm not sure if the function name is enough descriptive?